src/store/theme.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react';
+
+export const ThemeContext = React.createContext();
+
+export function withThemeContext(Element) {
+    return function WrapperComponent(props) {
+        return (
+            <ThemeContext.Consumer>
+                {state => (
+                    <Element
+                        {...props}
+                        context={{ ...state, ...props.context }}
+                    />
+                )}
+            </ThemeContext.Consumer>
+        );
+    };
+}
+
+class ThemeProvider extends Component {
+    constructor(props) {
+        super(props);
+
+        this.toggleTheme = () => {
+            this.setState(state => ({
+                theme: !state.theme
```

Comment:
If this is a booelan should be something like isDefaultTheme makes much more sense

Reply:
Sure, makes sense

Reply:
You resolved the discussion but didn't change the code